### ISSUE TITLE
IAliasAspect and the homepage

### ISSUE THREAD

**Issue by carlwoodhouse** [OrchardCMS/Orchard#6974]
I can't really decide if this is a bug myself, but it was a breaking change for us in 1.10.1 from 1.9.2 so might be worth some discussion.

Due to the way the homepage routing has been re-written IAliasAspect().Path no longer returns an empty path for the homepage, it returns the actual alias of the content item (not the HomeAlias)

For many people (myself included) it means we have to do some additional checking when using IAliasAspect to check if its the homepage as we dont want to surface multiple urls to the front end (for seo purposes)

Thoughts?

**Comment by sfmskywalker** [comment]
That's actually by design. Although arguably a poor one, it was introduced to fix a bug with export/import. The trouble was that in order for a content item to become the homepage, its Alias would be set to empty. This in itself is not bad, but the AutoroutePart uses it to build an identity. This meant that setting a content item to be the homepage meant changing its identity. In addition, it could be considered bad practice to change the URL of a content item (turning an existing routable content item into the homepage effectively changes its URL).

In any case, we're not too happy with the current solution either and are open to other suggestions.

**Comment by carlwoodhouse** [comment]
I assumed it was by design ;) 

Will think of some suggestions, but in terms of real-world use i also don't think its great that the homepage can now be served off two url's (i'd argue that promoting a contentitem to the homepage should at least redirect it's old URL to "/") - though that still leaves the current behaviour of IAliasAspect as undesirable imo.

**Comment by sfmskywalker** [comment]
I agree with that. Perhaps the AutoroutePart should be enhanced to support redirection, at least automatic redirection for when it is the homepage.

**Comment by carlwoodhouse** [comment]
I agree it should support redirection (in fact at Patient, we have a module for doing this for SEO purposes ... just preserves old aliases and if the thing they are pointed at doesn't match the DisplayAlias it redirects, which is actually how i noticed the change to the homepage alias lol)

But yeah, needs it at least for the homepage and i still think the aspect should really return "/" ;)
